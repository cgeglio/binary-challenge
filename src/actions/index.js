export const login = user => ({
  type: 'LOGIN',
  user
})

export const logout = user => ({
  type: 'LOGOUT',
  user
})

export const addCards = cards => ({
  type: 'ADD_CARDS',
  cards
})

export const removeCards = cards => ({
  type: 'REMOVE_CARDS',
  cards
})

export const addQuestion = question => ({
  type: 'ADD_QUESTION',
  question
})

export const removeQuestion = question => ({
  type: 'REMOVE_QUESTION',
  question
})

export const addFortune = fortune => ({
  type: 'ADD_FORTUNE',
  fortune
})

export const removeFortune = fortune => ({
  type: 'REMOVE_FORTUNE',
  fortune
})

export const addReading = currentReading => ({
  type: 'ADD_READING',
  currentReading
})

export const removeReading = currentReading => ({
  type: 'REMOVE_READING',
  currentReading
})

export const addFavorite = favorite => ({
  type: 'ADD_FAVORITE',
  favorite
})

export const removeFavorite = favorite => ({
  type: 'REMOVE_FAVORITE',
  favorite
})

export const resetFavorites = favorites => ({
  type: 'RESET_FAVORITES',
  favorites
})
