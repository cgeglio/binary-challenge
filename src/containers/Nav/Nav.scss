@import "../../mixins.scss";

.nav-bar {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 20px;
}

.nav-btn {
  @include buttons();
  height: 35px;
  margin: 0px 15px;
  width: 140px;
}

.nav-btn-container {
  align-self: baseline;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  margin: 30px 10px;
}

.logo-img {
  height: 100px;
  margin: 20px;
}
