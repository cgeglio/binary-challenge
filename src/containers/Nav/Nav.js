import React from 'react';
import { Link } from 'react-router-dom';
import './Nav.scss';
import { connect } from 'react-redux';
import { logout } from '../../actions';
import logo from '../../images/GoldenCity.png';

export const Nav = (props) => {
  return (
    <section className='nav-bar'>
      <img src={logo} alt="Golden City Tarot Logo" className='logo-img'/>
      <div className='nav-btn-container'>
        <Link to='/saved'><button className='nav-btn'>SAVED | {props.favorites.length || 0}</button></Link>
        {props.user.id ?
        <Link to='/'><button onClick={() => props.logoutUser(props.user)} id='logout-btn' className='nav-btn'>LOGOUT</button></Link>
        : <Link to='/'><button className='nav-btn'>LOGIN</button></Link>
        }
      </div>
    </section>
  )
}

export const mapStateToProps = state => ({
  user: state.user,
  favorites: state.favorites
})

export const mapDispatchToProps = dispatch => ({
  logoutUser: user => (dispatch(logout(user)))
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav)
