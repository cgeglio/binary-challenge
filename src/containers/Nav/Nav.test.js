import React from 'react';
import { Nav, mapStateToProps, mapDispatchToProps } from './Nav';
import { logout, resetFavorites, removeCards, removeReading, removeFortune, removeQuestion } from '../../actions';
import { shallow } from 'enzyme';

describe('Nav', () => {

  let wrapper;
  let mockUser;
  let mockFavorites;
  let mockCards;
  let mockQuestion;
  let mockFortune;
  let mockReading

  beforeEach(() => {
    mockUser = {name: 'Bithcuits', id: 50}
    mockFavorites = [{cards: [{name: 'The Magician', id: 28}]}];
    mockCards = [{name: 'The Moon'}];
    mockQuestion = 'Is the sky blue?';
    mockFortune = 'Two wrongs don\'t make a right'
    mockReading = {cards: [{name: 'The Magician', id: 28}], id: 30}
    wrapper = shallow(<Nav user={mockUser} favorites={mockFavorites} />);
  });

  describe('Nav container/component', () => {
    it ('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it ('should call logoutUser, resetFavorites, removeCards, removeQuestion, removeFortune, removeReading when the logout button is clicked', () => {
      let mockLogout = jest.fn().mockImplementation();
      let mockResetFavorites = jest.fn().mockImplementation();
      let mockRemoveCards = jest.fn().mockImplementation();
      let mockRemoveQuestion = jest.fn().mockImplementation();
      let mockRemoveFortune = jest.fn().mockImplementation();
      let mockRemoveReading = jest.fn().mockImplementation();
      wrapper = shallow(<Nav
        user={mockUser}
        favorites={mockFavorites}
        cards={mockCards}
        question={mockQuestion}
        fortune={mockFortune}
        currentReading={mockReading}
        logoutUser={mockLogout}
        resetFavorites={mockResetFavorites}
        removeCards={mockRemoveCards}
        removeQuestion={mockRemoveQuestion}
        removeFortune={mockRemoveFortune}
        removeReading={mockRemoveReading}
      />);
      wrapper.find('#logout-btn').simulate('click');
      expect(mockLogout).toHaveBeenCalledWith(mockUser);
      expect(mockResetFavorites).toHaveBeenCalledWith(mockFavorites);
      expect(mockRemoveCards).toHaveBeenCalledWith(mockCards);
      expect(mockRemoveQuestion).toHaveBeenCalledWith(mockQuestion);
      expect(mockRemoveFortune).toHaveBeenCalledWith(mockFortune);
      expect(mockRemoveReading).toHaveBeenCalledWith(mockReading);
    });
  });

  describe('mapStateToProps', () => {
    it('should return a user, an array of favorites, a question, a fortune, an array of cards, and a current reading', () => {
      const mockDispatch = jest.fn();
      const mockState = {
        user: mockUser,
        favorites: mockFavorites,
        question: mockQuestion,
        fortune: mockFortune,
        cards: mockCards,
        currentReading: mockReading,
        testKey: 'test value'
      }
      const expected = {
        user: mockUser,
        favorites: mockFavorites,
        question: mockQuestion,
        fortune: mockFortune,
        cards: mockCards,
        currentReading: mockReading
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    })
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the logout action when the logout button is clicked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = logout(mockUser);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.logoutUser(mockUser);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should call dispatch with the remove question action when the logout button is clicked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeQuestion(mockQuestion);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.removeQuestion(mockQuestion);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should call dispatch with the remove fortune action when the logout button is clicked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeFortune(mockFortune);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.removeFortune(mockFortune);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should call dispatch with the remove cards action when the logout button is clicked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeCards(mockCards);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.removeCards(mockCards);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should call dispatch with the remove reading action when the logout button is clicked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeReading(mockReading);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.removeReading(mockReading);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should call dispatch with the reset favorites action when the logout button is clicked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = resetFavorites(mockFavorites);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.resetFavorites(mockFavorites);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  });
});
