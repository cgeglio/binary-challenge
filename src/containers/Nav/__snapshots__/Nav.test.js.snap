// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav container/component should match the snapshot 1`] = `
<section
  className="nav-bar"
>
  <img
    alt="Golden City Tarot Logo"
    className="logo-img"
    src="GoldenCity.png"
  />
  <div
    className="nav-btn-container"
  >
    <Link
      to="/saved"
    >
      <button
        className="nav-btn"
      >
        SAVED | 
        1
      </button>
    </Link>
    <Link
      to="/"
    >
      <button
        className="nav-btn"
        id="logout-btn"
        onClick={[Function]}
      >
        LOGOUT
      </button>
    </Link>
  </div>
</section>
`;
